source/common/upstream/thread_aware_lb_impl.cc

```diff
@@ -12,9 +12,16 @@ namespace {
 void normalizeHostWeights(const HostVector& hosts, double normalized_locality_weight,
                           NormalizedHostWeightVector& normalized_host_weights,
                           double& min_normalized_weight, double& max_normalized_weight) {
-  uint32_t sum = 0;
+  // sum should be at most uint32_t max value, so we can validate it by accumulating into unit64_t
+  // and making sure there was no overflow
+  uint64_t sum = 0;
   for (const auto& host : hosts) {
     sum += host->weight();
+    if (sum > std::numeric_limits<uint32_t>::max()) {
```

Comment:
This seems like a fine way to check for overflow (the other one I could think of would be `if (sum + host->weight() >= sum)` with sum being uint32_t).

Can you make sure both this and the other exception has coverage and if not add a unit test?

Reply:
Thanks @snowp !
I agree that we can check if sum is increasing, but I think the current option clarifies that we check for an overflow.